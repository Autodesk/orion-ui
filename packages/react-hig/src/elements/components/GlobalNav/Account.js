/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import * as PropTypes from 'prop-types';
import HIGElement from '../../HIGElement';
import createComponent from '../../../adapters/createComponent';

export class Account extends HIGElement {
  commitUpdate(updatePayload, oldProps, newProp) {
    const mapping = {
      image: 'setImage',
      label: 'setLabel'
    };

    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case mapping[propKey]:
          this.hig[mapping[propKey]](propValue);
          break;
        case 'active':
          if (propValue) {
            this.hig.activate();
          } else {
            this.hig.deactivate();
          }
          break;          
        default:
          this.commitPropChange(propKey, propValue);
      }
    }
  }

  componentDidMount(){
    if (this.initialProps.activatedAccount == true ) {
      
    }
  }
}

const AccountComponent = createComponent(Account);

AccountComponent.propTypes = {
  image: PropTypes.string,
  label: PropTypes.string,
  activate: PropTypes.func,
  deactivate: PropTypes.func,
  onClick: PropTypes.func
};

AccountComponent.__docgenInfo = {
  props: {
    label: {
      description: 'sets {String} the label displayed for an account in Account/account switcher'
    },
    image: {
      description: 'sets {String} the image displayed for an account in Account/account switcher'
    },
    activate: {
      description: '{func} activates the Account'
    },
    deactivate: {
      description: '{func} deactivates the Account'
    },
    onclick: {
      description: '{func} calls the provided callback when user clicks on the Account'
    }
  }
};

export default AccountComponent;
