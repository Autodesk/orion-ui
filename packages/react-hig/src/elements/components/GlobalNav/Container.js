/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import HIGElement from '../../HIGElement';
import HIGChildValidator from '../../HIGChildValidator';
import createComponent from '../../../adapters/createComponent';

import TopNavComponent, { TopNav } from './TopNav';
import SubNavComponent, { SubNav } from './SubNav';
import Slot from './Slot';

export class Container extends HIGElement {
  componentDidMount() {
    if (this.topNav) {
      this.hig.addTopNav(this.topNav.hig);
      this.topNav.mount();
    }

    if (this.subNav) {
      this.hig.addSubNav(this.subNav.hig);
      this.subNav.mount();
    }

    if (this.slot) {
      this.hig.addSlot(this.slot);
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case TopNav:
        return new TopNav(this.hig.partials.TopNav, props);
      case SubNav:
        return new SubNav(this.hig.partials.SubNav, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance) {
    if (instance instanceof TopNav) {
      if (this.mounted) {
        this.hig.addTopNav(instance.hig);
        instance.mount();
      } else {
        this.topNav = instance;
      }
    } else if (instance instanceof SubNav) {
      if (this.mounted) {
        this.hig.addSubNav(instance.hig);
        instance.mount();
      } else {
        this.subNav = instance;
      }
    } else {
      throw new Error(`unknown type`);
    }
  }

  addSlot(element) {
    if (this.mounted) {
      this.hig.addSlot(element);
    } else {
      this.slot = element;
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    instance.unmount();
  }
}

const ContainerComponent = createComponent(Container);

ContainerComponent.propTypes = {
  children: HIGChildValidator([TopNavComponent, SubNavComponent, Slot])
};

ContainerComponent.__docgenInfo = {
  props: {
    children: {
      description: 'support adding TopNav, SubNav, and Slot'
    }
  }
};

ContainerComponent.TopNav = TopNavComponent;
ContainerComponent.SubNav = SubNavComponent;
ContainerComponent.Slot = Slot;

export default ContainerComponent;
