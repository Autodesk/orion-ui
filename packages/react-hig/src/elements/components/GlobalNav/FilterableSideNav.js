import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SideNav from '../../../adapters/GlobalNav/SideNav/SideNavAdapter';

const LinkList = SideNav.LinkList;
const Link = SideNav.LinkList.Link;
const Search = SideNav.Search;
const SectionList = SideNav.SectionList;
const Section = SideNav.SectionList.Section;
const SectionCollapse = SideNav.SectionList.Section.SectionCollapse;
const Group = SideNav.SectionList.Section.Group;
const Module = SideNav.SectionList.Section.Group.Module;
const ModuleCollapse = SideNav.SectionList.Section.Group.Module.ModuleCollapse;
const Submodule = SideNav.SectionList.Section.Group.Module.Submodule;

export default class FilterableSideNav extends Component {
  static propTypes = {
    query: PropTypes.string,
    items: PropTypes.shape({
      links: PropTypes.arrayOf(PropTypes.object),
      sections: PropTypes.arrayOf(
        PropTypes.shape({
          groups: PropTypes.arrayOf(
            PropTypes.shape({
              modules: PropTypes.shape({
                submodules: PropTypes.arrayOf(PropTypes.object)
              })
            })
          )
        })
      )
    }).isRequired
  };

  render() {
    return (
      <SideNav>
        {
          <LinkList>
            {this.props.items.links.map((link, i) => {
              return <Link {...link} key={link.title} />;
            })}
          </LinkList>
        }
      </SideNav>
    );
  }
}
