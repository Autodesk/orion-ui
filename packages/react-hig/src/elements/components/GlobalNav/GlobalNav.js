/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as HIG from 'hig.web';
import * as PropTypes from 'prop-types';

import HIGElement from '../../HIGElement';
import HIGChildValidator from '../../HIGChildValidator';
import createComponent from '../../../adapters/createComponent';

import SideNavComponent, { SideNav } from './SideNav';
import TopNavComponent, { TopNav } from './TopNav';
import SubNavComponent, { SubNav } from './SubNav';
import Slot from './Slot';

class GlobalNav extends HIGElement {
  constructor(initialProps) {
    super(HIG.GlobalNav, initialProps);
  }

  componentDidMount() {
    // Add any children
    if (this.sideNav) {
      this.hig.addSideNav(this.sideNav.hig);
      this.sideNav.componentDidMount();
    }

    if (this.topNav) {
      this.hig.addTopNav(this.topNav.hig);
      this.topNav.componentDidMount();
    }

    if (this.subNav) {
      this.hig.addSubNav(this.subNav.hig);
      this.subNav.componentDidMount();
    }

    if (this.initialProps.sideNavOpen) {
      this.hig.showSideNav();
    } else {
      this.hig.hideSideNav();
    }

    if (this.slot) {
      this.hig.addSlot(this.slot);
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case TopNav:
        return new TopNav(this.hig.partials.TopNav, props);
      case SubNav:
        return new SubNav(this.hig.partials.SubNav, props);
      case SideNav:
        return new SideNav(this.hig.partials.SideNav, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance, beforeChild = {}) {
    if (instance instanceof SideNav) {
      if (this.sideNav) {
        throw new Error('only one SideNav is allowed');
      } else {
        this.sideNav = instance;
        if (this.mounted) {
          this.hig.addSideNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else if (instance instanceof TopNav) {
      if (this.topNav) {
        throw new Error('only one TopNav is allowed');
      } else {
        this.topNav = instance;
        if (this.mounted) {
          this.hig.addTopNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else if (instance instanceof SubNav) {
      if (this.SubNav) {
        throw new Error('only one SubNav is allowed');
      } else {
        this.subNav = instance;
        if (this.mounted) {
          this.hig.addSubNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else {
      throw new Error('unknown type');
    }
  }

  addSlot(element) {
    if (this.mounted) {
      this.hig.addSlot(element);
    } else {
      this.slot = element;
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    if (instance instanceof SideNav) {
      this.sideNav = null;
    } else if (instance instanceof TopNav) {
      this.topNav = null;
    } else if (instance instanceof SubNav) {
      this.subNav = null;
    }
    instance.unmount();
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'sideNavOpen': {
          if (propValue) {
            this.hig.showSideNav();
          } else {
            this.hig.hideSideNav();
          }
          break;
        }
        case 'children': {
          /* no-op */
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const GlobalNavComponent = createComponent(GlobalNav);

GlobalNavComponent.propTypes = {
  sideNavOpen: PropTypes.bool,
  children: HIGChildValidator([
    SideNavComponent,
    TopNavComponent,
    SubNavComponent,
    Slot
  ])
};

GlobalNavComponent.__docgenInfo = {
  props: {
    sideNavOpen: {
      description: 'show or hide the SideNav'
    },

    children: {
      description: 'support adding SideNav, SubNav, or TopNav'
    }
  }
};

GlobalNavComponent.SideNav = SideNavComponent;
GlobalNavComponent.TopNav = TopNavComponent;
GlobalNavComponent.SubNav = SubNavComponent;
GlobalNavComponent.Slot = Slot;

export default GlobalNavComponent;
