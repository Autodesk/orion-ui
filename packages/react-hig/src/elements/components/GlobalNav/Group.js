/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';
import HIGNodeList from '../../HIGNodeList';
import HIGChildValidator from '../../HIGChildValidator';
import ModuleComponent, { Module } from './Module';

export class Group extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);
    this.modules = new HIGNodeList({
      type: Module,
      HIGConstructor: this.hig.partials.Module,
      onAdd: (instance, beforeInstance) => {
        this.hig.addModule(instance, beforeInstance);
      }
    });

    this.state = {};
    this._render = this._render.bind(this);
  }

  componentDidMount() {
    this.modules.componentDidMount();
    this._render();
  }

  createElement(ElementConstructor, props) {
    return this.modules.createElement(ElementConstructor, props);
  }

  insertBefore(instance, insertBeforeIndex) {
    this.modules.insertBefore(instance, insertBeforeIndex);
    if (this.mounted) {
      this._render();
    }
  }

  removeChild(instance) {
    this.modules.removeChild(instance);
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    this.processUpdateProps(updatePayload).then(this._render);
  }

  isVisible() {
    return this.state.isVisible;
  }

  expandModules() {
    this.modules.forEach(module => {
      module.commitUpdate({query: undefined})
      module.expandAll();
    });
    this._render();
  }

  collapseModules() {
    this.modules.forEach(module => {
      this.commitUpdate({query: undefined})
      module.collapseAll();
    });
    this._render();
  }

  _render() {
    const matches = this.modules.map(module => {
      module.commitUpdate({
        query: this.props.query,
        activeModule: this.props.activeModule,
        activeSubmodule: this.props.activeSubmodule
      });

      return module.isVisible();
    });

    if (matches.some(m => m)) {
      this.hig.show();
      this.state.isVisible = true;
    } else {
      this.hig.hide();
      this.state.isVisible = false;
    }
  }
}

const GroupComponent = createComponent(Group);

GroupComponent.propTypes = {
  children: HIGChildValidator([ModuleComponent])
};

GroupComponent.__docgenInfo = {
  props: {
    children: {
      description: 'support adding Module'
    }
  }
};

GroupComponent.Module = ModuleComponent;

export default GroupComponent;
