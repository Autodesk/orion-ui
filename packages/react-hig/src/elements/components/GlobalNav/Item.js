/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';

export class Item extends HIGElement {
  commitUpdate(updatePayload, oldProps, newProps) {
    const mapping = {
      icon: 'setIcon',
      title: 'setTitle',
      link: 'setLink'
    };

    this.commitUpdateWithMapping(updatePayload, mapping)
  }
}

const ItemComponent = createComponent(Item);

ItemComponent.propTypes = {
  icon: PropTypes.string,
  title: PropTypes.string,
  link: PropTypes.string,
  onClick: PropTypes.func,
  onHover: PropTypes.func
};

ItemComponent.__docgenInfo = {
  props: {
    icon: {
      description: 'sets the icon of an item'
    },

    title: {
      description: 'sets the title of an item'
    },

    link: {
      description: 'sets the link of an item'
    },

    onClick: {
      description: 'triggered when a link is clicked on'
    },

    onHover: {
      description: 'triggered when a link is hovered over'
    }
  }
};

export default ItemComponent;
