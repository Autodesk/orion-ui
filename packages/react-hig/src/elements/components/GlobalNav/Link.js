/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';

export class Link extends HIGElement {
  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'title':
          this.hig.setTitle(propValue);
          break;
        case 'link':
          this.hig.setLink(propValue);
          break;
        case 'onClick': {
          const dispose = this._disposeFunctions.get('onClickDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onClickDispose',
            this.hig.onClick(propValue)
          );
          break;
        }
        case 'onHover': {
          const dispose = this._disposeFunctions.get('onHoverDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onHoverDispose',
            this.hig.onClick(propValue)
          );
          break;
        }

        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const LinkComponent = createComponent(Link);

export default LinkComponent;
