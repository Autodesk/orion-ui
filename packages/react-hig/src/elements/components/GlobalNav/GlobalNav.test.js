/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import { mount } from 'enzyme';
import * as HIG from 'hig.web';
import React from 'react';

import GlobalNav from './GlobalNav';

describe('<GlobalNav>', () => {
  /**
     * Creates a hig.web GlobalNav and returns the instance and it's container
     */
  function createHigNav() {
    const higContainer = document.createElement('div');

    // use spread here to clone defaults since HIG.Button mutates this object
    const higNav = new HIG.GlobalNav();

    higNav.mount(higContainer);

    return { higNav, higContainer };
  }

  it('renders the global nav', () => {
    const { higNav, higContainer } = createHigNav();

    const reactContainer = document.createElement('div');
    const wrapper = mount(<GlobalNav />, { attachTo: reactContainer });

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

    expect(reactContainer.firstChild.outerHTML).toEqual(
      higContainer.firstChild.outerHTML
    );
  });

  describe('sideNavOpen prop', () => {
    it('shows an error if it is not a boolean', () => {
      global.console.error = jest.fn();

      mount(<GlobalNav sideNavOpen="foo" />);

      expect(console.error).toBeCalledWith(
        expect.stringMatching(
          /Invalid prop `sideNavOpen` of type `string` supplied to `GlobalNav`, expected `boolean`./
        )
      );
    });

    it('is equal to showSideNav', () => {
      const { higNav, higContainer } = createHigNav();
      const reactContainer = document.createElement('div');
      const wrapper = mount(<GlobalNav />, { attachTo: reactContainer });

      higNav.showSideNav();

      wrapper.setProps({ sideNavOpen: true });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );

      higNav.hideSideNav();
      wrapper.setProps({ sideNavOpen: false });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('can showSideNav by default', () => {
      const { higNav, higContainer } = createHigNav();
      const reactContainer = document.createElement('div');
      const wrapper = mount(<GlobalNav sideNavOpen={true} />, {
        attachTo: reactContainer
      });

      higNav.showSideNav();

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('can render the SideNav as a child', () => {
      const { higNav, higContainer } = createHigNav();

      const sideNav = new higNav.partials.SideNav();
      higNav.addSideNav(sideNav);

      const reactContainer = document.createElement('div');
      const wrapper = mount(
        <GlobalNav>
          <GlobalNav.SideNav />
        </GlobalNav>,
        {
          attachTo: reactContainer
        }
      );

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('can only render a single SideNav', () => {
      expect(() => {
        mount(
          <GlobalNav>
            <GlobalNav.SideNav />
            <GlobalNav.SideNav />
          </GlobalNav>
        );
      }).toThrowError(/only one SideNav is allowed/);
    });
  });

  it('can render the Container as a child', () => {
    const { higNav, higContainer } = createHigNav();

    const reactContainer = document.createElement('div');
    const wrapper = mount(
      <GlobalNav>
      </GlobalNav>,
      {
        attachTo: reactContainer
      }
    );

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

    expect(reactContainer.firstChild.outerHTML).toEqual(
      higContainer.firstChild.outerHTML
    );
  });

  it('can not render arbitrary HTML elements as children', () => {
    global.console.error = jest.fn();

    mount(
      <GlobalNav>
        <GlobalNav.TopNav />
        <div>Hello world!</div>
      </GlobalNav>
    );

    expect(console.error).toBeCalledWith(
      expect.stringMatching(
        /'div' is not a valid child of GlobalNav. Children should be of type 'SideNav, TopNav, SubNav, Slot'/
      )
    );
  });

  it('can not render text elements as children', () => {
    global.console.error = jest.fn();

    mount(
      <GlobalNav>
        some random child text
      </GlobalNav>
    );

    expect(console.error).toBeCalledWith(
      expect.stringMatching(
        /'some random child text' is not a valid child of GlobalNav. Children should be of type 'SideNav, TopNav, SubNav, Slot'/
      )
    );
  });
});
