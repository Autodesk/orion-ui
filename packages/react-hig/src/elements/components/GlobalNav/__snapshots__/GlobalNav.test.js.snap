// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalNav> can render the Container as a child 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<GlobalNav> renders the global nav 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<GlobalNav> sideNavOpen prop can render the SideNav as a child 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__sidenav__scroll\\">
        <!--SEARCH-->
        <!--SECTION-->
        <div class=\\"hig__global-nav__sidenav__links\\">
            <!--LINK-->
        </div>
        <div class=\\"hig__global-nav__sidenav__copyright\\">Â© 2017 Autodesk, Inc.</div>
    </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<GlobalNav> sideNavOpen prop can showSideNav by default 1`] = `
"<div class=\\"hig__global-nav hig__global-nav--open\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<GlobalNav> sideNavOpen prop is equal to showSideNav 1`] = `
"<div class=\\"hig__global-nav hig__global-nav--open\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<GlobalNav> sideNavOpen prop is equal to showSideNav 2`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;
